api/src/services/submission-job-queue-service.ts

```diff
@@ -0,0 +1,112 @@
+import { IDBConnection } from '../database/db';
+import { SubmissionJobQueueRepository } from '../repositories/submission-job-queue-repository';
+import { SUBMISSION_MESSAGE_TYPE, SUBMISSION_STATUS_TYPE } from '../repositories/submission-repository';
+import { generateDatasetS3FileKey, uploadFileToS3 } from '../utils/file-utils';
+import { DBService } from './db-service';
+import { SubmissionService } from './submission-service';
+
+export interface IProprietaryInformation {
+  first_nations_id: number;
+  proprietor_type_id: number;
+  survey_id: number;
+  rational: string;
+  proprietor_name: number;
+  disa_required?: boolean;
+}
+
+export class SubmissionJobQueueService extends DBService {
+  repository: SubmissionJobQueueRepository;
+
+  constructor(connection: IDBConnection) {
+    super(connection);
+
+    this.repository = new SubmissionJobQueueRepository(connection);
+  }
+
+  /**
+   * Intakes DwCA and preps it for processing. Adding files to S3, tracks submission status and creates a queue record
+   *
+   * @param {string} dataUUID
+   * @param {Express.Multer.File} file
+   * @return {*}  {Promise<number>}
+   * @memberof SubmissionJobQueueService
+   */
+  async intake(
+    dataUUID: string,
+    file: Express.Multer.File,
+    proprietaryInformation?: IProprietaryInformation
+  ): Promise<{ queue_id: number }> {
+    const submissionService = new SubmissionService(this.connection);
+    const nextJobId = await this.repository.getNextQueueId();
+
+    const key = await this.uploadDatasetToS3(dataUUID, nextJobId.queueId, file);
+
+    const submission = await submissionService.getSubmissionIdByUUID(dataUUID);
```

Comment:
Instead of `let submissionId`, could we `let submission` and replace it if needed the in the subsequent create method?

Reply:
done